src/NovelTextRect.h

```diff
@@ -7,41 +7,42 @@
 #include <string>
 #include "GraphicsCharacterRenderData.h"
 #include "NovelImageRect.h"
+#include "ShaderProgram.h"
 
 namespace NovelRT {
-  class NovelTextRect : public NovelRenderObject {
-
-  private:
-    RGBAConfig _colourConfig;
-    std::string _fontFileDir;
-    std::string _previousFontFileDir = "";
-    float _fontSize;
-    std::map<GLchar, GraphicsCharacterRenderData> _fontCharacters;
-    std::string _text = "";
-    std::vector<NovelImageRect*> _letterRects;
-    NovelCommonArgs _args;
-
-    void reloadText();
-
-  protected:
-    void configureObjectBuffers() final;
-
-  public:
-    NovelTextRect(NovelLayeringService* layeringService,
-                  float fontSize,
-                  float screenScale,
-                  const std::string& fontFileDir,
-                  const RGBAConfig& colourConfig,
-                  const NovelCommonArgs& args,
-                  GLuint programId);
-
-    RGBAConfig getColourConfig() const;
-    void setColourConfig(const RGBAConfig& value);
-
-    std::string getText() const;
-    void setText(const std::string& value);
-
-    void drawObject() final;
-  };
+class NovelTextRect : public NovelRenderObject {
+private:
+  void reloadText();
+
+  RGBAConfig _colourConfig;
```

Comment:
nit: might be nice to sort the order on some of these from largest to smallest as it will help with packing/layout